### PR TITLE
[NCL-4958] BPM Refactor

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by michalovjan** [comment]
I have no idea what I've done

File: bpm/src/main/java/org/jboss/pnc/bpm/model/mapper/BuildResultMapper.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import static java.util.Optional.ofNullable;
+
+@ApplicationScoped
+public class BuildResultMapper {
+
+    @Inject
+    RepositoryManagerResultMapper repositoryManagerResultMapper;
```
Comment (janinko) [inline bpm/src/main/java/org/jboss/pnc/bpm/model/mapper/BuildResultMapper.java]: private

File: bpm/src/main/java/org/jboss/pnc/bpm/model/mapper/RepositoryManagerResultMapper.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RepositoryManagerResultMapper {
+
+    @Inject
+    ArtifactMapper artifactMapper;
```
Comment (janinko) [inline bpm/src/main/java/org/jboss/pnc/bpm/model/mapper/RepositoryManagerResultMapper.java]: private

File: bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultSerializationTest.java [hunk 3]
```diff
@@ -16,40 +16,93 @@
...
+    Configuration configuration;
+
+    @Spy
+    TargetRepositoryMapper targetRepositoryMapper;
+
+    @Spy
+    AbstractArtifactMapperImpl artifactMapper;
+
+    @Spy
+    RepositoryManagerResultMapper repositoryManagerResultMapper;// = new RepositoryManagerResultMapper();
```
Comment (janinko) [inline bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultSerializationTest.java]: comment

File: bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultSerializationTest.java [hunk 4]
```diff
@@ -16,40 +16,93 @@
...
+
+    @Spy
+    @InjectMocks
+    BuildResultMapper buildResultMapper;// = new BuildResultMapper();
```
Comment (janinko) [inline bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultSerializationTest.java]: comment

File: bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultSerializationTest.java [hunk 5]
```diff
@@ -16,40 +16,93 @@
...
 /**
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
+@RunWith(MockitoJUnitRunner.class)
 public class BuildResultSerializationTest {
 
     private final Logger log = LoggerFactory.getLogger(BuildResultSerializationTest.class);
 
+    @Mock
+    Configuration configuration;
```
Comment (janinko) [inline bpm/src/test/java/org/jboss/pnc/bpm/test/BuildResultSerializationTest.java]: private?

File: bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java [hunk 6]
```diff
@@ -16,41 +16,91 @@
...
  * Test serialization of Repository manager rest
  */
+@RunWith(MockitoJUnitRunner.class)
 public class RepositoryManagerResultSerializationTest {
 
     private final Logger log = LoggerFactory.getLogger(RepositoryManagerResultSerializationTest.class);
 
+
+    @Mock
+    Configuration configuration;
```
Comment (janinko) [inline bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java]: private?

File: bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java [hunk 7]
```diff
@@ -16,41 +16,91 @@
...
+        mapper.registerModule(new JavaTimeModule());
 
         RepositoryManagerResult repoMngrResult = RepositoryManagerResultMock.mockResult();
-        RepositoryManagerResultRest repoMngrResultRest = new RepositoryManagerResultRest(repoMngrResult);
+        RepositoryManagerResultRest repoMngrResultRest = repositoryManagerResultMapper.toDTO(repoMngrResult);
 
         String repoMngrResultJson = JsonOutputConverterMapper.apply(repoMngrResultRest);
         log.debug("RepoManagerResultJson : {}", repoMngrResultJson);
 
         RepositoryManagerResultRest deserializedRepoManResultRest = mapper.readValue(repoMngrResultJson, RepositoryManagerResultRest.class);
```
Comment (janinko) [inline bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java]: Maybe replace with JsonOutputConverterMapper?

File: bpm/src/main/java/org/jboss/pnc/bpm/model/RepositoryManagerResultRest.java [hunk 8]
```diff
@@ -16,43 +16,36 @@
...
 /**
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
+@AllArgsConstructor
 @XmlRootElement(name = "repositoryManagerResult")
 public class RepositoryManagerResultRest implements Serializable {
-    private List<ArtifactRest> builtArtifacts;
-    private List<ArtifactRest> dependencies;
+    private List<org.jboss.pnc.dto.Artifact> builtArtifacts;
+    private List<org.jboss.pnc.dto.Artifact> dependencies;
```
Comment (janinko) [inline bpm/src/main/java/org/jboss/pnc/bpm/model/RepositoryManagerResultRest.java]: is JIRA for verification/fix in maitai created?

File: bpm/src/main/java/org/jboss/pnc/bpm/model/BuildExecutionConfigurationRest.java [hunk 9]
```diff
@@ -133,7 +160,8 @@ public BuildExecutionConfiguration toBuildExecutionConfiguration() {
         );
     }
 
-    public BuildConfigurationAuditedRest createBuildConfigurationAuditedRest() {
+    //TODO was this used? couldn't find anything
```
Comment (janinko) [inline bpm/src/main/java/org/jboss/pnc/bpm/model/BuildExecutionConfigurationRest.java]: so remove it?

File: bpm/src/main/java/org/jboss/pnc/bpm/model/RepositoryManagerResultRest.java (hunk 8 above)
Comment (michalovjan) [reply bpm/src/main/java/org/jboss/pnc/bpm/model/RepositoryManagerResultRest.java]: On it!

File: bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java (hunk 7 above)
Comment (michalovjan) [reply bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java]: Could you elaborate? 
Do you mean to use:
`JsonOutputConverterMapper.apply(repositoryManagerResultMapper.toDTO(repoMngrResult))`

File: bpm/src/main/java/org/jboss/pnc/bpm/model/BuildExecutionConfigurationRest.java (hunk 9 above)
Comment (michalovjan) [reply bpm/src/main/java/org/jboss/pnc/bpm/model/BuildExecutionConfigurationRest.java]: FINE

File: bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java (hunk 7 above)
Comment (janinko) [reply bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java]: `JsonOutputConverterMapper.readValue(repoMngrResultJson, RepositoryManagerResultRest.class);`

File: bpm/src/main/java/org/jboss/pnc/bpm/model/RepositoryManagerResultRest.java (hunk 8 above)
Comment (michalovjan) [reply bpm/src/main/java/org/jboss/pnc/bpm/model/RepositoryManagerResultRest.java]: NCL-5052 created

File: bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java (hunk 7 above)
Comment (michalovjan) [reply bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java]: Oh you mean to delete ObjectMapper!

**Comment by michalovjan** [comment]
Let there be conflicts!!

**Comment by thescouser89** [comment]
michal++

**Pull request merged**